ibm/data_source_ibm_is_virtual_endpoint_gateway.go

```diff
@@ -122,6 +121,7 @@ func dataSourceIBMISEndpointGatewayRead(
 	allrecs := []vpcv1.EndpointGateway{}
 	for {
```

Comment:
i think we can remove for loop in all these changed datasources as only one instance of the resource will be returned with name filter